apps/prairielearn/src/lib/github.js

```diff
@@ -50,25 +50,30 @@ async function createRepoFromTemplateAsync(client, repo, template) {
 
   // The above call will complete before the repo itself is actually ready to use,
   // so poll for a bit until all the files are finally copied in.
-  let repo_up = false;
-  const poll_time_ms = 100;
-  while (!repo_up) {
+  //
+  // We'll try for up to 30 seconds, polling every second.
+  for (let i = 0; i < 30; i++) {
     try {
       // If the repo is not ready yet, this will fail with "repo is empty"
       await client.repos.getContent({
         owner: config.githubCourseOwner,
         repo: repo,
-        path: '/',
+        path: '',
       });
-      repo_up = true;
+      return;
     } catch (err) {
-      logger.debug(`${repo} is not ready yet, polling again in ${poll_time_ms} ms`);
-    }
+      // We'll get a 404 if the repo is not ready yet. If we get any other
+      // error, then something unexpected happened and we should bail out.
+      if (err.status !== 404) {
+        throw err;
+      }
```

Comment:
This will allow us to fail quickly if e.g. we're rate limited or if we get an unexpected 400/500 like above.